python/tvm/relay/op/strategy/x86.py

```diff
@@ -84,8 +88,13 @@ def conv2d_strategy_cpu(attrs, inputs, out_type, target):
         raise ValueError("dilation should be positive value")
 
     if groups == 1:
-        if layout == "NCHW":
-            assert kernel_layout == "OIHW"
+        if layout.startswith("NCHW"):
```

Comment:
This is unnecessary because after alter op layout pass, the `conv2d` op will be convert into `conv2d_NCHWc` op and will go into `conv2d_NCHWc_strategy_cpu` function.

Reply:
Actually the changes here are necessary.  `_alter_conv2d_layout` which is used in `AlterOpLayout` pass will call `select_implementation(relay.op.get('nn.conv2d'))`  which will finally go to this function `conv2d_strategy_cpu`. And the layout is passed as blocking layout

Reply:
No, inside `_alter_conv2d_layout`, the `conv2d` layout should still be NCHW, and this function will then convert layout into NCHWc

Reply:
Could you separate "NCHW" and "NCHWc" into two different if branches?

Reply:
Ok, changed. Extract the same part of code handling HCWH and NCHWc to an nested function.